drivers/pppos.c

```diff
@@ -511,9 +513,13 @@ static void pppos_mainLoop(void* _state)
 		}
 #endif
 
-		if ((res = at_send_cmd(state->fd, AT_CONNECT_CMD, AT_CONNECT_CMD_TIMEOUT_MS)) != AT_RESULT_CONNECT) {
-			log_warn("failed to dial PPP, res=%d, retrying", res);
-			goto fail;
+		retry = AT_CONNECT_CMD_TIMEOUT_MS / PPPOS_CONNECT_CMD_RETRY_MS;
+		while ((res = at_send_cmd(state->fd, AT_CONNECT_CMD, AT_CONNECT_CMD_TIMEOUT_MS)) != AT_RESULT_CONNECT) {
+			if (retry-- <= 0 || res < 0) {
```

Comment:
I would add at least comment that res will be < 0 in case timeout happened inside `at_send_cmd` - and/or **why** we would need to repeat the `at_send_cmd` at all (the generalized description of the pull request).

My initial thought was that the "old" modem was not handled properly (that we would retry the `at_send_cmd` X times even if each of the call would take `AT_CONNECT_CMD_TIMEOUT_MS`). I needed to check the implementation of `at_send_cmd` to understand the code - which should not be the case.